lib/zip_source_win32a.c

```diff
@@ -111,7 +111,17 @@ static int
 _win32_rename_temp_a(_zip_source_win32_read_file_t *ctx)
 {
     if (!MoveFileExA(ctx->tmpname, ctx->fname, MOVEFILE_REPLACE_EXISTING))
-	return -1;
+        return -1;
+
+    DWORD attributes = GetFileAttributesA(ctx->fname);
+    if (!attributes)
```

Comment:
Thanks for the fixes. One more thing: `GetFileAttributesA` and `GetFileAttributesW` return `INVALID_FILE_ATTRIBUTES` on errors (see https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-getfileattributesa) so the check should compare against that, and not zero. Please fix this, then I'll merge.